env.go

```diff
@@ -282,7 +292,7 @@ func getOr(key, defaultValue string, defExists bool, envs map[string]string) (va
 	switch {
 	case (!exists || key == "") && defExists:
 		return defaultValue, true
-	case !exists:
+	case !exists || key == "":
```

Comment:
This would've hidden the issue - this bug results in an env variable with the key "" and the value "::=::\" being set, so it breaks nested struct parsing, which relies on getting "" from `get`:
```
		value, err := get(refTypeField, opts)
		if err != nil {
			return err
		}
		if value == "" {
			if reflect.Struct == refField.Kind() {
				if err := doParse(refField, funcMap, opts); err != nil {
					return err
				}
			}
			continue
		}
```